### PR TITLE
Windows support

### LINKED ISSUE #148: Make repo work with node 6.5.0
Please create a wiki or elaborate on the installation guide for windows users following the readme when I type `npm i` my system tells me that it cannont find npm at all. However `nvm install 5` also installs `npm` so this error makes no sense to me.

### PR DESCRIPTION
**Pull request by robinjoseph08**
◝(๑⁺᷄д⁺᷅๑)◞՞

fixes #148 

after much trial and error, this repo finally runs on windows! yay! this pr makes changes to make it work on windows and on node v6 and it makes docs changes to ensure that support isnt lost by accident

@ami you can just checkout the branch, `rm -rf node_modules`, `npm i`, and make sure everything still runs to test it

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -7,3 +7,11 @@ Some guidelines you should follow when submitting a PR:
...
 * Before submitting the PR, you should make sure that your code passes the linter. Running `npm run lint` locally will speed things up because the TravisCI build will fail if the linting doesn't pass.
+
+## Shrinkwrap
+
+Since this project is built using Node and NPM, we have a `package.json` to list all of the top-level dependencies that we use. Since the version numbers in there can leave a lot up for interpretation, we also have an [`npm-shrinkwrap.json` file](https://docs.npmjs.com/cli/shrinkwrap) to lock down exact versions of direct dependencies and the versions of nested dependencies. One issue with the shrinkwrap though is that [it can make any optional dependencies required](https://github.com/npm/npm/issues/2679). One such example of this is the [`fsevents`](https://www.npmjs.com/package/fsevents) module. This is an optional dependencies that is only able to be build on Mac OS X. So if this module becomes required, `npm i` on any machine other than a Mac will fail.
```
Comment (ami) [inline CONTRIBUTING.md]: "This is an optional dependec**ies**" ➡️  "This is an optional dependenc**y**"

Comment (ami) [inline CONTRIBUTING.md]: "**build** on Mac OS X" ➡️ "**built** on Mac OS X"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -7,3 +7,11 @@ Some guidelines you should follow when submitting a PR:
...
+
+Since there currently isn't a fix for this just yet, we've employed [a workaround](https://github.com/npm/npm/issues/2679#issuecomment-150084700) of just deleting the optional dependency from the shrinkwrap and then manually listing it under `optionalDependencies`. The problem with this is that while it's easy to take it out, it's difficult to keep it out. Because TravisCI doesn't support running a build on Windows to ensure functionality, we have to get a little manual and just make sure `fsevents` isn't included in the shrinkwrap. Even though this isn't going to be exhaustive, it's at least a start.
```
Comment (ami) [inline CONTRIBUTING.md]: maybe italicize "take" and "keep"? i had to re-read a few times to see what you meant.

**Comment by ami** [comment]
lgtm after wording changes!

**Comment by robinjoseph08** [comment]
@ami updated `CONTRIBUTING.md` with your changes! and i also updated `README.md` to make a note about using nvm since we've seen people having problems with it. could you take a look at it too? :3

**Pull request merged**